### PR TITLE
fix: ensure that Rook Data will be cleanup when rook is removed

### PR DESCRIPTION
**Pull request by camilamacedo86**
#### What this PR does / why we need it:

Currently, when we are removing Rook we are not removing the data. 
That can cause problems mainly if Rook be re-installed on the cluster. 
Also, uses space in disk which is no longer required. 

#### Which issue(s) this PR fixes:

Fixes # [sc-70238]

#### Special notes for your reviewer:
See that we are also adding the check to the testgrid

#### Does this PR introduce a user-facing change?

```release-note
Fixes Rook data not been removed when Rook Ceph is removed from the cluster.
```

### CODE REVIEW

File: testgrid/specs/full.yaml [hunk 1]
```diff
@@ -1841,4 +1845,9 @@
...
     else
        echo "Namespace rook-ceph was removed"
-    fi
\ No newline at end of file
+    fi
+    echo "Verify if Rook data was removed"
+    if [ -d "/var/lib/rook" ] || [ -d "/opt/replicated/rook" ]; then
+        logWarn  "Data within /var/lib/rook, /opt/replicated/rook and any bound disks has not been freed."
+    fi
+
```
Comment (camilamacedo86) [inline testgrid/specs/full.yaml:1853]:
```suggestion

```

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: scripts/common/rook.sh [hunk 2]
```diff
@@ -116,6 +120,13 @@ function remove_rook_ceph() {
...
+    rm -rf /var/lib/rook || true
+    rm -rf /opt/replicated/rook || true
+
+    if [ -d "/var/lib/rook" ] || [ -d "/opt/replicated/rook" ]; then
+        logWarn  "Data within /var/lib/rook, /opt/replicated/rook and any bound disks has not been freed."
+    fi
+
     # print success message
     printf "%bRemoved rook-ceph successfully!\n%b" "$GREEN" "$NC"
     printf "Data within /var/lib/rook, /opt/replicated/rook and any bound disks has not been freed.\n"
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply scripts/common/rook.sh]: Yes, we should.

**Comment by emosbaugh** [comment]
I will wait for [this pr](https://github.com/replicatedhq/kURL/pull/4209) to be merged before reviewing.

File: scripts/common/rook.sh [hunk 3]
```diff
@@ -133,9 +138,15 @@ function remove_rook_ceph() {
         kubectl -n kurl scale deploy ekc-operator --replicas=1
     fi
 
+    rm -rf /var/lib/rook || true
+    rm -rf /opt/replicated/rook || true
```
Comment (emosbaugh) [inline scripts/common/rook.sh:142]: I think this looks good but I'd like @laverya 's opinion as to whether he feels it is safe to remove the data here if potentially other operations failed above with only a logWarn

Comment (camilamacedo86) [reply scripts/common/rook.sh:142]: From the docs; https://rook.io/docs/rook/v1.10/Getting-Started/ceph-teardown/#delete-the-operator-and-related-resources

Screenshot 2023-03-20 at 21 58 05

Also, @laverya comment that would make sense we remove here: https://app.shortcut.com/replicated/story/70238/migration-from-rook-does-not-clean-the-data-stored-and-are-not-following-the-rook-teardown-docs#:~:text=Mar%2010%202023%2C%205%3A11%20pm

Comment (emosbaugh) [reply scripts/common/rook.sh:142]: yes. im just worried that there are plenty of conditionals above here that do not return when a command fails and instead log warn and continue. is it safe to delete the directories if we have not yet successfully removed rook and ceph?

Comment (laverya) [reply scripts/common/rook.sh:142]: I think to get here the namespace has to have been successfully deleted, so it's fine to delete the hostpaths

Comment (camilamacedo86) [reply scripts/common/rook.sh:142]: We only are in the place where we delete the data after:
a) we migrate 
b) we delete the namespace

The path : https://github.com/replicatedhq/kURL/pull/4223/files#diff-d1a3d4d46c348ae45d17405ff5db5f2a57f6de8264635200a462fb8a7ac9b7bdR91-R93

Will also allow the Operator remove the data. 
That seems sorted out the scenarios to remove Rook for upper versions either.

File: scripts/common/rook.sh [hunk 4]
```diff
@@ -81,11 +81,17 @@ function remove_rook_ceph() {
...
     fi
+
     log "Removing rook-ceph Storage Classes"
     if ! kubectl get storageclass | grep rook | awk '{ print $1 }' | xargs -I'{}' kubectl delete storageclass '{}' --timeout=60s; then
         logFail "Unable to delete rook-ceph StorageClasses"
         return 1
     fi
+
+    # More info: https://rook.io/docs/rook/v1.10/Getting-Started/ceph-teardown/#delete-the-cephcluster-crd
+    log "Allow RookCeph delete the data"
```
Comment (laverya) [inline scripts/common/rook.sh:92]:
```suggestion
    log "Patch Ceph cluster to allow deletion"
```

File: scripts/common/rook.sh (hunk 3 above)
Comment (emosbaugh) [reply scripts/common/rook.sh:142]: sounds good. approving

**Review by emosbaugh — APPROVED** [review]

**Comment by camilamacedo86** [comment]
Moving forward since we got via the slack (private)  the OK from @laverya as well.

**Pull request merged**